cli/command_repository_connect.go

```diff
@@ -66,9 +66,12 @@ func (c *connectOptions) setup(svc appServices, cmd *kingpin.CmdClause) {
 	// Set up flags shared between 'create' and 'connect'. Note that because those flags are used by both command
 	// we must use *Var() methods, otherwise one of the commands would always get default flag values.
 	cmd.Flag("cache-directory", "Cache directory").PlaceHolder("PATH").Envar(svc.EnvName("KOPIA_CACHE_DIRECTORY")).StringVar(&c.connectCacheDirectory)
-	cmd.Flag("content-cache-size-mb", "Size of local content cache").PlaceHolder("MB").Default("5000").Int64Var(&c.connectMaxCacheSizeMB)
-	cmd.Flag("metadata-cache-size-mb", "Size of local metadata cache").PlaceHolder("MB").Default("5000").Int64Var(&c.connectMaxMetadataCacheSizeMB)
-	cmd.Flag("max-list-cache-duration", "Duration of index cache").Default("30s").Hidden().DurationVar(&c.connectMaxListCacheDuration)
+
+	c.maxListCacheDuration = 30 * time.Second //nolint:gomnd
+	c.contentCacheSizeMB = 5000
+	c.maxMetadataCacheSizeMB = 5000
+	c.cacheSizeFlags.setup(cmd)
```

Comment:
Instead of setting the defaults inline in all places, I think if you can pass the defaults to the setup() function as parameters which will then set it up on the CLI. That way, even the CLI help will see the defaults for connect/create/set commands.